app/views/ForwardLivechatView.tsx

```diff
@@ -54,22 +54,33 @@ interface IForwardLivechatViewProps {
 	forwardRoom: (rid: string, transferData: ITransferData) => void;
 }
 
+const COUNT_DEPARTMENT = 50;
+
 const ForwardLivechatView = ({ forwardRoom, navigation, route, theme }: IForwardLivechatViewProps) => {
 	const [departments, setDepartments] = useState<IParsedData[]>([]);
 	const [departmentId, setDepartment] = useState('');
+	const [departmentTotal, setDepartmentTotal] = useState(0);
+	const [departmentOffset, setDepartmentOffset] = useState(0);
 	const [users, setUsers] = useState<IParsedData[]>([]);
 	const [userId, setUser] = useState();
 	const [room, setRoom] = useState<IRoom>({});
 
 	const rid = route.params?.rid;
 
-	const getDepartments = async () => {
+	const getDepartments = async (text = '', offset = 0) => {
 		try {
-			const result = await RocketChat.getDepartments();
+			const result = await RocketChat.getDepartments({ count: COUNT_DEPARTMENT, text, offset });
 			if (result.success) {
-				setDepartments(
-					result.departments.map((department: ILivechatDepartment) => ({ label: department.name, value: department._id }))
-				);
+				const parsedDepartments = result.departments.map((department: ILivechatDepartment) => ({
+					label: department.name,
+					value: department._id
+				}));
+				if (!text && !offset) {
```

Comment:
I insert this to save the departments only once when the text is empty and offset is 0, instead of all time that this function is called.